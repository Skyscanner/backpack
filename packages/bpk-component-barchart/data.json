{
  "prices": [
    {
      "month": "Jan",
      "price": 305
    },
    {
      "month": "Feb",
      "price": 348
    },
    {
      "month": "Mar",
      "price": 418
    },
    {
      "month": "Apr",
      "price": 448
    }
  ]
}
