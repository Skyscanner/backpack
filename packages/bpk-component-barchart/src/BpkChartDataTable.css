@keyframes bpk-keyframe-spin{100%{transform:rotate(1turn)}}.bpk-chart-data-table{position:absolute;width:1px;height:1px;margin:-1px;padding:0;border:0;overflow:hidden;clip:rect(0 0 0 0)}.bpk-chart-data-table thead,.bpk-chart-data-table tbody{position:absolute;width:1px;height:1px;margin:-1px;padding:0;border:0;overflow:hidden;clip:rect(0 0 0 0)}
