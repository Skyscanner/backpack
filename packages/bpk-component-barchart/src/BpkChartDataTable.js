/*
 * Backpack - Skyscanner's Design System
 *
 * Copyright 2016 Skyscanner Ltd
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/* @flow strict */

import PropTypes from 'prop-types';
import React from 'react';

import { cssModules } from '../../bpk-react-utils';

import dataProp from './customPropTypes';
import STYLES from './BpkChartDataTable.module.scss';

const getClassName = cssModules(STYLES);

type Props = {
  data: Array<any>, // We pass any here as the array can contain free form data depending on the user
  xScaleDataKey: string,
  yScaleDataKey: string,
  xAxisLabel: string,
  yAxisLabel: string,
};

const BpkChartDataTable = (props: Props) => {
  const { data, xAxisLabel, xScaleDataKey, yAxisLabel, yScaleDataKey } = props;
  const rows = data.map((point, i) => {
    const key = `chart-data-table-row-${i}`;

    return (
      <tr key={key}>
        <td>{point[xScaleDataKey]}</td>
        <td>{point[yScaleDataKey]}</td>
      </tr>
    );
  });

  return (
    <table className={getClassName('bpk-chart-data-table')}>
      <thead>
        <tr>
          <th>{xAxisLabel}</th>
          <th>{yAxisLabel}</th>
        </tr>
      </thead>
      <tbody>{rows}</tbody>
    </table>
  );
};

BpkChartDataTable.propTypes = {
  data: dataProp, // This is actually required and will fail if not present
  xScaleDataKey: PropTypes.string.isRequired,
  yScaleDataKey: PropTypes.string.isRequired,
  xAxisLabel: PropTypes.string.isRequired,
  yAxisLabel: PropTypes.string.isRequired,
};

BpkChartDataTable.defaultProps = {
  data: null,
};

export default BpkChartDataTable;
