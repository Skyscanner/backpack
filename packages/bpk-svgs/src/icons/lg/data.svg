<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20.883 10c.596 0 1.064-.52.926-1.1a9.076 9.076 0 0 0-6.71-6.71C14.52 2.054 14 2.52 14 3.118V9a1 1 0 0 0 1 1h5.883z"/><path d="M20 12c.552 0 1.006.45.945.998a9.001 9.001 0 1 1-9.943-9.943c.549-.06.998.393.998.945v6a2 2 0 0 0 2 2h6z"/></svg>