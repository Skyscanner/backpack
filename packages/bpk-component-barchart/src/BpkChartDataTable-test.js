/*
 * Backpack - Skyscanner's Design System
 *
 * Copyright 2016-2020 Skyscanner Ltd
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/* @flow strict */

import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import data from '../data.json';

import BpkChartDataTable from './BpkChartDataTable';

const { prices } = data;

describe('BpkChartDataTable', () => {
  it('should render correctly', () => {
    const tree = shallow(
      <BpkChartDataTable
        xScaleDataKey="month"
        yScaleDataKey="price"
        xAxisLabel="Month"
        yAxisLabel="Average price (Â£)"
        data={prices}
      />,
    );
    expect(toJson(tree)).toMatchSnapshot();
  });
});
