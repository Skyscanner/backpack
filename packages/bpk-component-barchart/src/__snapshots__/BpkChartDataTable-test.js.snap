// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BpkChartDataTable should render correctly 1`] = `
<DocumentFragment>
  <table
    class="bpk-chart-data-table"
  >
    <thead>
      <tr>
        <th>
          Month
        </th>
        <th>
          Average price (Â£)
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Jan
        </td>
        <td>
          305
        </td>
      </tr>
      <tr>
        <td>
          Feb
        </td>
        <td>
          348
        </td>
      </tr>
      <tr>
        <td>
          Mar
        </td>
        <td>
          418
        </td>
      </tr>
      <tr>
        <td>
          Apr
        </td>
        <td>
          448
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
